examples/3_simple_storage/simple_storage.go

```diff
@@ -0,0 +1,231 @@
+security-conscious bot development with persistent state. It is a
+place to store small bits of data that are
+  (1) encrypted for a team or user (via the user's implicit self-team: e.g.
+alice,alice),
+  (2) persistent across logins
+  (3) fast and durable.
+
+It supports putting, getting, listing, and deleting. There is also
+concurrency support, but check out 5_secret_storage for that. A team has many
+namespaces, a namespace has many entryKeys, and an entryKey has one current
+entryValue. Namespaces and entryKeys are in cleartext, and the Keybase client
+service will encrypt and sign the entryValue on the way in (as well as
+decrypt and verify on the way out) so keybase servers cannot see it or
+forge it.
+
+This example shows how you can use interact with the team
+encrypted key-value store using chat commands.
+*/
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/keybase/go-keybase-chat-bot/kbchat"
+	"github.com/keybase/go-keybase-chat-bot/kbchat/types/chat1"
+)
+
+func fail(msg string, args ...interface{}) {
+	fmt.Fprintf(os.Stderr, msg+"\n", args...)
+	os.Exit(3)
+}
+
+/*
+ Listens to chat messages of the form:
+    `!storage put <namespace> <key> <value> (<revision>)`
+    `!storage get <namespace> <key>`
+    `!storage delete <namespace> <key> (<revision>)`
+    `!storage list`  // list namespaces
+    `!storage list <namespace>`  // list entries in namespace
+*/
+func main() {
+	const MsgPrefix = "!storage"
+	const (
+		Help   = "help"
+		List   = "list"
+		Get    = "get"
+		Put    = "put"
+		Delete = "delete"
+	)
+
+	var kbLoc string
+	var kbc *kbchat.API
+	var err error
+
+	flag.StringVar(&kbLoc, "keybase", "keybase", "the location of the Keybase app")
+	flag.Parse()
+
+	fmt.Println("Simple kvstore storage bot now starting...")
+
+	if kbc, err = kbchat.Start(kbchat.RunOptions{KeybaseLocation: kbLoc}); err != nil {
+		fail("Error creating API: %s", err.Error())
+	}
+
+	sub, err := kbc.ListenForNewTextMessages()
+	if err != nil {
+		fail("Error listening: %s", err.Error())
+	}
+
+	fmt.Println("Now listening for new chat messages...")
+	for {
+		msg, err := sub.Read()
+
+		if err != nil {
+			fail("failed to read message: %s", err.Error())
+		}
+
+		if msg.Message.Content.TypeName != "text" {
+			continue
+		}
+
+		channel := msg.Message.Channel
+		team := channel.Name
+		body := strings.Split(strings.TrimSpace(msg.Message.Content.Text.Body), " ")
+
+		if len(body) < 2 || body[0] != MsgPrefix {
+			continue
+		}
+
+		switch action := body[1]; action {
+		case Help:
+			handleHelp(kbc, channel, team, body, action)
+		case List:
+			handleList(kbc, channel, team, body, action)
+		case Get:
+			handleGet(kbc, channel, team, body, action)
+		case Put:
+			handlePut(kbc, channel, team, body, action)
+		case Delete:
+			handleDelete(kbc, channel, team, body, action)
+		}
+	}
+}
+
+func handleHelp(api *kbchat.API, channel chat1.ChatChannel, team string, body []string, action string) {
```

Comment:
Could have an advertisement in addition or instead of the help message. Here's an example: https://github.com/mmaxim/bottender/blob/master/bottender.go#L49-L96

Reply:
neat!